### PR TITLE
Check order of arguments in `orient_axis()`

### LINKED ISSUE #21659: Creating a vector raises a deprecation warning and need better error message
Looks like a deprecation warning for sympify is now appearing with sympy mechanics objects. Also, we need a better error message if you accidentally swap the order of arguments in `orient_axis()` (or its companion methods).

```
In [2]: import sympy.physics.mechanics as me

In [3]: theta = me.dynamicsymbols('theta')

In [4]: N = ReferenceFrame('N')
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-4-563293f48d8e> in <module>
----> 1 N = ReferenceFrame('N')

NameError: name 'ReferenceFrame' is not defined

In [5]: N = me.ReferenceFrame('N')

In [6]: A = me.ReferenceFrame('A')

In [7]: A.orient_axis(N, theta, N.x + N.y + N.z)
/home/moorepants/miniconda/lib/python3.7/site-packages/sympy/core/sympify.py:462: SymPyDeprecationWarning: 

String fallback in sympify has been deprecated since SymPy 1.6. Use
sympify(str(obj)) or sympy.core.sympify.converter or obj._sympy_
instead. See https://github.com/sympy/sympy/issues/18066 for more
info.

  deprecated_since_version='1.6'
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
ValueError: Error from parse_expr with transformed code: 'N .x +N .y +N .z '

The above exception was the direct cause of the following exception:

AttributeError                            Traceback (most recent call last)
<ipython-input-7-28003ca07dd8> in <module>
----> 1 A.orient_axis(N, theta, N.x + N.y + N.z)

~/miniconda/lib/python3.7/site-packages/sympy/physics/vector/frame.py in orient_axis(self, parent, axis, angle)
    611         _check_frame(parent)
    612 
--> 613         amount = sympify(angle)
    614         theta = amount
    615         axis = _check_vector(axis)

~/miniconda/lib/python3.7/site-packages/sympy/core/sympify.py in sympify(a, locals, convert_xor, strict, rational, evaluate)
    477     try:
    478         a = a.replace('\n', '')
--> 479         expr = parse_expr(a, local_dict=locals, transformations=transformations, evaluate=evaluate)
    480     except (TokenError, SyntaxError) as exc:
    481         raise SympifyError('could not parse %r' % a, exc)

~/miniconda/lib/python3.7/site-packages/sympy/parsing/sympy_parser.py in parse_expr(s, local_dict, transformations, global_dict, evaluate)
   1016         return eval_expr(code, local_dict, global_dict)
   1017     except Exception as e:
-> 1018         raise e from ValueError(f"Error from parse_expr with transformed code: {code!r}")
   1019 
   1020 

~/miniconda/lib/python3.7/site-packages/sympy/parsing/sympy_parser.py in parse_expr(s, local_dict, transformations, global_dict, evaluate)
   1014 
   1015     try:
-> 1016         return eval_expr(code, local_dict, global_dict)
   1017     except Exception as e:
   1018         raise e from ValueError(f"Error from parse_expr with transformed code: {code!r}")

~/miniconda/lib/python3.7/site-packages/sympy/parsing/sympy_parser.py in eval_expr(code, local_dict, global_dict)
    909     """
    910     expr = eval(
--> 911         code, global_dict, local_dict)  # take local objects in preference
    912     return expr
    913 

<string> in <module>

AttributeError: 'function' object has no attribute 'x'
```

### PR DESCRIPTION
**Pull request by sidhu1012**
#### References to other Issues or PRs

Fixes #21659

#### Release Notes

* physics.vector
    * The `orient_axis()` works if the order of axis and angle is swapped.

### CODE REVIEW

**Comment by sidhu1012** [comment]
Now it is

```
>>> N = ReferenceFrame('N')
>>> A = ReferenceFrame('A')
>>> theta = dynamicsymbols('theta')
>>> A.orient_axis(N, theta, N.x + N.y + N.z)

Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/root/.vscode-server/extensions/ms-python.python-2021.6.944021595/pythonFiles/lib/python/debugpy/__main__.py", line 45, in <module>
    cli.main()
  File "/root/.vscode-server/extensions/ms-python.python-2021.6.944021595/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 444, in main
    run()
  File "/root/.vscode-server/extensions/ms-python.python-2021.6.944021595/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 285, in run_file
    runpy.run_path(target_as_str, run_name=compat.force_str("__main__"))
  File "/usr/lib/python3.8/runpy.py", line 265, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/usr/lib/python3.8/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/root/sympy/sympy/physics/vector/tests/test_frame.py", line 556, in <module>
    A.orient_axis(N, theta, N.x + N.y + N.z)
  File "/root/sympy/sympy/physics/vector/frame.py", line 649, in orient_axis
    axis = _check_vector(axis)
  File "/root/sympy/sympy/physics/vector/vector.py", line 807, in _check_vector
    raise TypeError('A Vector must be supplied')
TypeError: A Vector must be supplied
```

**Comment by sidhu1012** [comment]
If a user do something like-

```
>>> N = ReferenceFrame('N')
>>> A = ReferenceFrame('A')
>>> theta = dynamicsymbols('theta')
>>> A.orient_axis(N, N.x + N.y + N.z, N.x)
```
then https://github.com/sympy/sympy/issues/21659#issue-929915143 error would be displayed, which seems right now as vectors are not sympifiable and `theta` must be sympifiable.

@moorepants Would you like me to make it strict? This way seems better though.

**Comment by moorepants** [comment]
This is still a very poor error message "TypeError: A Vector must be supplied". Ideally the error message would inform the user of the argment order. At the minimum "A vector must be supplied in the second argument". Or "The arguments are orient_axis(frame, angle, axis)". The user gets the current error message and did supply a vector, just not in the right order.

**Comment by sidhu1012** [comment]
@moorepants I think the user can always check order in documentation, we shouldn't check order in code, we should just check if the terms passed are correct according to the correct order.

**Comment by moorepants** [comment]
> @moorepants I think the user can always check order in documentation, we shouldn't check order in code, we should just check if the terms passed are correct according to the correct order.

I disagree. If I had to guess about 50% of the time I guess the order correctly. The 50% bad guesses gives me the non-useful error. An ideal error message would say "you need to swap the order of the vector and the scalar" or something similar. Then I don't have to go digging around in the documentation.

**Comment by sidhu1012** [comment]
@moorepants Now the error message is `The order of arguments is incorrect. The arguments are orient_axis(frame, axis, angle)`

**Comment by sidhu1012** [comment]
@moorepants It's ready to be merged.

**Comment by smichr** [comment]
When there are three different, distinguishable inputs, why but use a star Arg and sort it out and raise an error if a type is missing?

**Comment by moorepants** [comment]
Yeh, we could do that, but we wrote this new method to be very explicit about what to pass in. So that would take us backwards a bit. One confounding issue here is that we made a mistake on the order compared to the earlier similar methods.

**Comment by moorepants** [comment]
@smichr Are you suggesting that we just slyly swap the args to the correct order internally?

I guess we could leave the explicit arg order here, but internally allow the vector and scalar to be swapped. Then we have no error message, we just fix it if the user passes it in wrong. We can leave it undocumented (or have a note that this will occur). I think that would be safe in this case and shouldn't cause any issues.

**Comment by sidhu1012** [comment]
Yes I like it more than giving error.

File: sympy/physics/vector/tests/test_frame.py [hunk 1]
```diff
@@ -454,6 +454,8 @@ def test_orient_axis():
     A = ReferenceFrame('A')
     B = ReferenceFrame('B')
     assert A.orient_axis(B,-B.x, 1) == A.orient_axis(B, B.x, -1)
+    assert A.orient_axis(B,-B.x, 1) == A.orient_axis(B, 1, -B.x)
```
Comment (moorepants) [inline sympy/physics/vector/tests/test_frame.py:457]: These two functions do not return anything, so the result here is `assert None == None`, right? I don't think this is what you are hoping for in this test.

Comment (sidhu1012) [reply sympy/physics/vector/tests/test_frame.py]: I corrected that

**Comment by moorepants** [comment]
LGTM, automerge is on. Thanks!

**Comment by smichr** [comment]
> swap the args to the correct order internally

yes. In a way, `range` does this when it juggles the three possible arguments to mean min, max and step. It's different, though. But the idea of "set of non-ambiguous args" seems useful especially if one type could never have the representation of another. e.g. in geometry if a method needed a Segment and a Point but a Segment could collapse to a Point if the end-points are the same then it could be ambiguous which was to be the segment and which the point. So if the input is unambiguous then the intent is obvious, "what I mean, know you."

**Comment by moorepants** [comment]
Ok, I'm good with that. Thanks for the suggestion.

**Pull request merged**